src/read.rs

```diff
@@ -511,6 +543,27 @@ impl<R: Read + Seek> ZipArchive<R> {
             search_upper_bound,
         )?;
         search_results.into_iter().for_each(|(footer64, archive_offset)| {
+            let archive_offset = match config.archive_offset {
+                ArchiveOffset::None => 0,
```

Comment:
Check this before doing the search on line 540, or at least add a parameter to find_and_parse that lets it stop after 1 valid result, to make sure we aren't checking the same offset more than once.

Reply:
Yeah we still need to find the footer, so I can add something to make it stop after one result.

Reply:
Hmm actually, technically we don't know whether the first footer found actually corresponds to the archive offset passed (i.e. it could be junk), so I think it probably should work the way it currently works here?